pkg/openlambda/ol.go

```diff
@@ -53,10 +62,57 @@ func NewConfig(logger srk.Logger, config *viper.Viper) (srk.FunctionService, err
 		isLocal:        isLocal,
 		sessionStarted: false,
 		log:            logger,
+		stats:          olStats{0, 0},
 	}
 	return olCfg, nil
 }
 
+func (self *olConfig) ReportStats(reset bool) (map[string]float64, error) {
+
+	if !self.sessionStarted {
+		return nil, errors.New("No active OpenLambda sessions")
+	}
+
+	olResp, err := http.Post(self.urls[0]+"/stats", "application/json", strings.NewReader(""))
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to POST stats request to ol worker")
+	}
+
+	respBuf, err := ioutil.ReadAll(olResp.Body)
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to read stats response")
+	}
+
+	stats := make(map[string]float64)
+
+	var rawDecoded interface{}
+	if err = json.Unmarshal(respBuf, &rawDecoded); err != nil {
+		return nil, errors.Wrap(err, "Failed to interpret OL statistics")
+	}
+
+	decoded := rawDecoded.(map[string]interface{})
+	for k, v := range decoded {
+		if floatv, ok := v.(float64); ok {
+			stats[k] = floatv
+		} else {
+			self.log.Warnf("Ignoring non-numeric statistics result from openLambda: %v=%v", k, v)
+		}
+	}
+	stats["srkInvoke"] = self.stats.tInvoke / float64(self.stats.nInvoke)
+
+	if reset {
+		//Reset the statistics
+		_, err = http.Post(self.urls[0]+"/stats", "application/json", strings.NewReader("reset"))
```

Comment:
If we're going to be making two calls to the API, one to get the stats and one to reset, then why do we have both under the same `ReportStats(bool)` interface? Should we consider `ReportStats()` and `ResetStats()` instead?

Reply:
I agree. I've switched to separate report/reset.